argilla/src/argilla/settings/_io/_hub.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import httpx
+import warnings
+from collections import defaultdict
+from enum import Enum
+from typing import Any, Dict, TYPE_CHECKING, Union, List, Optional
+
+from argilla._exceptions._hub import DatasetsServerException
+from argilla._exceptions._settings import SettingsError
+
+if TYPE_CHECKING:
+    from argilla import Settings
+
+DATASETS_SERVER_BASE_URL = "https://datasets-server.huggingface.co"
+DATASETS_SERVER_HEADERS = {"Accept": "application/json"}
+DATASETS_SERVER_TIMEOUT = 30
```

Comment:
do we have back-off retry policy somewhere else by chance?

Reply:
Not that I know of, for the datasets server API.